ui/src/components/friends/add.rs

```diff
@@ -146,27 +146,11 @@ pub fn AddFriend(cx: Scope) -> Element {
         }
     });
 
-    let id_ch = use_coroutine(cx, |mut rx: UnboundedReceiver<()>| {
-        to_owned![my_id];
-        async move {
-            let warp_cmd_tx = WARP_CMD_CH.tx.clone();
-            while rx.next().await.is_some() {
-                let (tx, rx) = oneshot::channel::<Result<DID, warp::error::Error>>();
-                if let Err(e) =
-                    warp_cmd_tx.send(WarpCmd::MultiPass(MultiPassCmd::GetOwnDid { rsp: tx }))
-                {
-                    log::error!("failed to send warp command: {}", e);
-                    continue;
-                }
+    let mut did_short = "#".to_string();
+    did_short.push_str(&state.read().get_own_identity().short_id().to_string());
```

Comment:
it might be better to do something like

```rust
let identity = state.read().get_own_identity();
let username_with_short_id = format!("{}#{}", identity.username(), identity.short_id());
```

This would reduce the redundant calls.